### DISCUSSION TITLE
Suggestion for patch releases

### DISCUSSION THREAD

**Discussion by gh-andre** [MengRao/fmtlog#38] [General]
It is helpful when patch releases are focused on just the fix and don't bring other changes with them. For example, the fix for `FMTLOG_BLOCK` was layered on top of other changes, like this.
```
git log --format="%h %d %s" v2.1.0~1..v2.1.2
bc3dfe4  (tag: v2.1.2) fix an bug about FMTLOG_BLOCK
8ba235c  Update README.md
6553081  Update README.md
8aa7fd8  make format string type consistent with fmt::format/print, and stop using FMT_STRING
c8ce0e3  relax some use of noexcept
d7e66ee  fix an issue on copy aligned objects
c87ebeb  add noexcept to functions
686df5a  (tag: v2.1.1) fix a compiling error
1d92ad9  (tag: v2.1.0) optimize latency and code size; upgrade fmtlib to 8.1.0
```
You could tag a dot-zero release, as you do today, but if a patch-level fix is required, you could make a branch for patches for that release. These fixes would also be cherry-picked into `main`. It would look like this:
```
               v2.1.1               v2.1.2
         +--------x--------------------y---------> v2.1.x
        / compiler error fix   FMTLOG_BLOCK fix
       /
 -----o------x----o---o----o---y----o------------> main
   v2.1.0
```
The fixes `x` and `y` are cherry-picked to `main`. The patch branch never merges into `main` - it's there only to provide patch-level fixes.

This would allow people take patch fixes without worrying about accompanying possibly-incompatible changes.

**Comment by MengRao** [comment]
Thanks, I'll take it into consideration in future releases :)
